pkg/cachemanager/cachemanager.go

```diff
@@ -171,15 +175,25 @@ func (c *CacheManager) replaceWatchSet(ctx context.Context) error {
 	if err != nil {
 		// account for any watches failing to remove
 		if f := watch.NewErrorList(); errors.As(err, &f) && !f.HasGeneralErr() {
-			c.handleDanglingWatches(watch.SetFrom(f.RemoveGVKFailures()))
+			removeGVKFailures := watch.SetFrom(f.RemoveGVKFailures())
+			finallyRemoved := c.danglingWatches.Difference(removeGVKFailures)
+
+			c.gvksToDeleteFromCache.AddSet(finallyRemoved)
+			c.danglingWatches.RemoveSet(finallyRemoved)
+			c.danglingWatches.AddSet(removeGVKFailures.Difference(c.danglingWatches))
```

Comment:
Just add the set directly, not the difference between the two sets. The extra layer of indirection lowers readability and costs compute while being logically equivalent.